src/pages/index.js

```diff
@@ -61,16 +107,64 @@ const addImageModal = new ModalWithForm(
 //-----------CREATE A MODALWITHIMAGE INSTANCE FOR THE PREVIEW IMAGE------------>>
 const previewModal = new ModalWithImage("#image-preview-modal");
 
+//-----------CREATE A FUNCTION THAT HANDLES DELETE CLICK------------>>
+function handleDeleteClick(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setCallback(() => {
+    api.deleteCard(card.getId()).catch((err) => console.log(err));
+    card.deleteCard();
+    deleteConfirmationModal.close();
+  });
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES LIKE CLICK------------>>
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
+    .then((res) => card.toggleLikeCard(res.isLiked))
+    .catch((err) => console.log(err));
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES AVATAR EDIT SUBMIT------------>>
+function handleAvatarFormSubmit(values) {
+  avatarEditModal.renderSaving(true);
+  api
+    .changeAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+    })
+    .catch((err) => console.log(err))
+    .finally(avatarEditModal.renderSaving(false));
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 

**Please, correct it everywhere**